local-tests/chainrunner/node.py

```diff
@@ -0,0 +1,92 @@
+import json
+import os.path as op
+import re
+import requests
+import jsonrpcclient as rpc
+import subprocess
+
+from .utils import flags_from_dict
+
+class node:
+    """A class representing a single node of a running blockchain."""
+    def __init__(self, binary, chainspec, path, logdir=None):
+        self.chainspec = chainspec
+        self.binary = binary
+        self.path = path
+        self.logdir = logdir or path
+        self.logfile = None
+        self.flags = {}
+        self.running = False
+
+
+    def _stdargs(self):
+        return ['--base-path', self.path, '--chain', self.chainspec]
+
+
+    def start(self, name):
```

Comment:
a general, also controversial recommendation - I know that a big part of Python's power comes from no need of typing, but still I am a fan of type hints in public API - it makes the interface much cleaner and transparent, also some IDEs can use them to perform better analysis, and you don't have to specify type in function name (like `flags_from_dict` could be then renamed to `flags_from(d: Dict)`)